library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/async/RecordedDataQueueHandlerTest.kt

```diff
@@ -0,0 +1,455 @@
+
+        val spyTimestamp = spy.timestamp
+        whenever(mockTimeProvider.getDeviceTimestamp())
+            .thenReturn(spyTimestamp + MAX_DELAY_MS + 1)
+
+        // When
+        testedHandler.tryToConsumeItems()
+        mockExecutorService.shutdown()
+        mockExecutorService.awaitTermination(1, TimeUnit.SECONDS)
+
+        // Then
+        assertThat(testedHandler.recordedDataQueue.isEmpty()).isTrue()
+        verifyNoMoreInteractions(mockProcessor)
+    }
+
+    @Test
+    fun `M remove item from queue W tryToConsumeItems() { invalid item }`() {
+        // Given
+        val spy = spy(fakeSnapshotQueueItem)
+        spy.nodes = emptyList()
+        spy.systemInformation = mockSystemInformation
+        doReturn(false).whenever(spy).isValid()
+
+        testedHandler.recordedDataQueue.offer(spy)
+
+        val spyTimestamp = spy.timestamp
+        whenever(mockTimeProvider.getDeviceTimestamp())
+            .thenReturn(spyTimestamp)
+
+        // When
+
+        testedHandler.tryToConsumeItems()
+        mockExecutorService.shutdown()
+        mockExecutorService.awaitTermination(1, TimeUnit.SECONDS)
+
+        // Then
+        assertThat(testedHandler.recordedDataQueue.size).isEqualTo(0)
+        verifyNoMoreInteractions(mockProcessor)
+    }
+
+    @Test
+    fun `M do nothing W tryToConsumeItems() { item not ready }`() {
+        // Given
+        val spy = spy(fakeSnapshotQueueItem)
+
+        doReturn(true).whenever(spy).isValid()
+        doReturn(false).whenever(spy).isReady()
+
+        testedHandler.recordedDataQueue.add(spy)
+
+        whenever(mockTimeProvider.getDeviceTimestamp())
+            .thenReturn(fakeRumContextData.timestamp)
+
+        // When
+        testedHandler.tryToConsumeItems()
+        mockExecutorService.shutdown()
+        mockExecutorService.awaitTermination(1, TimeUnit.SECONDS)
+
+        // Then
+        verifyNoMoreInteractions(mockProcessor)
+    }
+
+    @Test
+    fun `M call processor W tryToConsumeItems() { valid snapshot item }`() {
+        // Given
+        val spy = spy(fakeSnapshotQueueItem)
+
+        doReturn(true).whenever(spy).isValid()
+        doReturn(true).whenever(spy).isReady()
+
+        testedHandler.recordedDataQueue.add(spy)
+        val spyTimestamp = spy.timestamp
+
+        whenever(mockTimeProvider.getDeviceTimestamp())
+            .thenReturn(spyTimestamp)
+
+        // When
+        testedHandler.tryToConsumeItems()
+        mockExecutorService.shutdown()
+        mockExecutorService.awaitTermination(1, TimeUnit.SECONDS)
+
+        // Then
+        verify(mockProcessor).processScreenSnapshots(
+            nodes = spy.nodes,
+            systemInformation = spy.systemInformation,
+            newContext = spy.newRumContext,
+            prevContext = spy.prevRumContext,
+            timestamp = spy.timestamp
+        )
+    }
+
+    @Test
+    fun `M call processor W tryToConsumeItems() { valid Touch Event item }`() {
```

Comment:
I understand that this way you can assert the actual item in the `mockProcessor` but I would also find useful to test this logic when calling the actual public methods:
`addSnapshotItem` or `addTouchEventItem` even if you cannot assert the actual item you could test that: 
```
verify(mockProcessor).processTouchEventRecords(argumentCaptor<SessionReplayContext>.capture(), argumentCaptor<List<MobileSegment.MobileRecord>>().capture())
```
And then you can assert the value in those `argumentCaptors`.
Same thing for the above test for `processScreenSnapshots`